work_queue/src/work_queue_process.c

```diff
@@ -267,6 +267,37 @@ pid_t work_queue_process_execute(struct work_queue_process *p )
 			close(p->output_fd);
 			exit(0);
 			*/
+			
+			/*
+			struct list *infiles = p->task->input_files; 
+			char *file = (char *)list_pop_head(infiles);
+			printf("file: %d", list_size(p->task->input_files));
+			*/
+
+			// this is returning a data from f->payload, but says that there is no such file when I try to open it
+			struct work_queue_file *f;
+			list_first_item(p->task->input_files);
+			while((f = list_next_item(p->task->input_files))) {
+				char *file_name = f->payload;
```

Comment:
I think you want t->remote_name here, instead of t->payload.  (See work_queue_internal.h)

Reply:
That worked. The latest commit is fully working reading data from input file and sending to network function. It didn't look like there was anything in `dttools`for this so I used `fread` but if I missed something I can change that

Reply:
I also tested with different formats of JSON on the manger end and modifying the handler in the network function accordingly and that worked, so the JSON schema doesn't matter at all in terms of the worker, which is what we're looking for